subcommands/keys/tuf_utils.go

```diff
@@ -212,44 +212,81 @@ func GetOfflineCreds(credsFile string) (OfflineCreds, error) {
 	return files, nil
 }
 
-func FindTufSigner(keyid, pubkey string, creds OfflineCreds) (*TufSigner, error) {
-	pubkey = strings.TrimSpace(pubkey)
-	for k, v := range creds {
-		if strings.HasSuffix(k, ".pub") {
-			tk := client.AtsKey{}
-			if err := json.Unmarshal(v, &tk); err != nil {
-				return nil, fmt.Errorf("Unable to parse JSON for %s: %w", k, err)
-			}
-			if strings.TrimSpace(tk.KeyValue.Public) == pubkey {
-				pkname := strings.Replace(k, ".pub", ".sec", 1)
-				pkbytes := creds[pkname]
-				tk = client.AtsKey{}
-				if err := json.Unmarshal(pkbytes, &tk); err != nil {
-					return nil, fmt.Errorf("Unable to parse JSON for %s: %w", pkname, err)
-				}
-				keyType, err := parseTufKeyType(tk.KeyType)
-				if err != nil {
-					return nil, fmt.Errorf("Unsupported key type for %s: %s", pkname, tk.KeyType)
-				}
-				pk, err := keyType.ParseKey(tk.KeyValue.Private)
-				if err != nil {
-					return nil, fmt.Errorf("Unable to parse key value for %s: %w", pkname, err)
-				}
-				return &TufSigner{
-					Id:   keyid,
-					Type: keyType,
-					Key:  pk,
-				}, nil
-			}
+func FindOneTufSigner(root *client.AtsTufRoot, creds OfflineCreds, keyids []string) (signer TufSigner, err error) {
+	var signers []TufSigner
+	if signers, err = findTufSigners(root, creds, keyids); err == nil {
+		if len(signers) == 0 {
+			err = fmt.Errorf("Found no active signing key for: %v.", keyids)
+		} else if len(signers) > 1 {
+			err = fmt.Errorf(`Found more than one active signing key for: %v.
+This is an unsupported and insecure way to store private keys.
```

Comment:
This is a functional change in behavior, right? not against it, but probably need to mention this in the next release notes.

Reply:
An answer to your question is a bit more complex:
- From the current implementation perspective, this is a forward-looking non-functional change. We only allow to have 1 active offline TUF root key and 1 active offline TUF targets key at a time. Thus, there is no way this condition can be true.
- At the same time, this is a change defines a future function of the multi-key TUF support, in accordance to our agreed design.

So, I'd say that this is a regular _new feature_, which is fully backward-compatible.
Therefore it needs not be mentioned explicitly in release notes for the next fioctl version, other than a part of a sentence about the _multi-key TUF support_.